packages/spartan/extra/gen_resolvconf.py

```diff
@@ -69,7 +69,20 @@ def check_server(addr):
 
 # If Spartan is not up, fall back, and insert the upstreams
 else:
-    fallback_servers = os.environ['RESOLVERS'].split(',')
+    fallback_servers = []
+
+    # Resolvconf does not support custom ports, skip if not default
+    for ns in os.environ['RESOLVERS'].split(','):
+        ip, separator, port = ns.rpartition(':')
+        if not separator:
+            fallback_servers.append(ns)
+            continue
+        if port == "53":
+            fallback_servers.append(ip)
```

Comment:
@cmaloney Not sure what you mean:

```
>>> host, sep, port = '192.168.1.1'.rpartition(':')
>>> not sep
True
>>> host, sep, port = '192.168.1.1:53'.rpartition(':')
>>> not sep
False
```

Are you just suggesting a stricter check on separator being an empty string or populated?